src/test/unit/profile-upgrading.test.js

```diff
@@ -304,3 +304,27 @@ describe('upgrading gecko profiles', function() {
     expect(geckoProfile12).toEqual(afterUpgradeGeckoReference);
   });
 });
+
+describe('importing perf profile', function() {
+  it('should import a perf profile', function() {
+    let version = -1;
+    try {
+      const fs = require('fs');
+      const zlib = require('zlib');
+      const buffer = fs.readFileSync('src/test/fixtures/upgrades/test.perf.gz');
+      const decompressedArrayBuffer = zlib.gunzipSync(buffer);
+      const text = decompressedArrayBuffer.toString('utf8');
+      const profile = unserializeProfileOfArbitraryFormat(text);
+      if (profile === undefined) {
+        throw new Error('Unable to parse the profile.');
+      }
+      //console.log(profile);
+      version = profile.meta.version;
+      expect(profile).toMatchSnapshot();
```

Comment:
I'm a bit concerned about the size of this test fixture. It generates an 100,000+ line snapshot. Generally snapshots are meant to be somewhat human readable, and when they fail they give nice diffs. This one seems to be a bit too long to work nicely. In addition, the test takes `1144ms` to run on my machine, which isn't the slowest test we have, but this could be reduced with a smaller fixture file.

If you'd like to use a large input file, perhaps we could also gzip the resulting gecko profile? That way it won't show up in our greps when we're working on files.

Reply:
made it much smaller